internals/garbage-collector.rst

```diff
@@ -319,10 +319,15 @@ the same object survives another GC round in this new generation (generation 1)
 it will be moved to the last generation (generation 2) where it will be
 surveyed the least often.
 
-Generations are collected when the number of objects that they contain reaches some
-predefined threshold, which is unique for each generation and is lower the older
-the generations are. These thresholds can be examined using the  ``gc.get_threshold``
-function:
+In order to decide when to run, the collector keeps track of the number object
+allocations and deallocations since the last collection. When the number of
+allocations minus the number of deallocations exceeds ``threshold_0``,
+collection starts. Initially only generation 0 is examined. If generation 0 has
+been examined more than ``threshold_`` times since generation 1 has been
+examined, then generation 1 is examined as well. With the third generation,
```

Comment:
Does "the third generation" refer to "generation 2" here? Or what happens to the second generation? At least to me, this seems like this could be a bit confusing...

Reply:
We can say "generation 2", but I would say is clear from the previous number that the generation 0 is the first and generation 1 is the 2nd

Reply:
Perhaps I'm just a n00b, but I sure was more than a little confused by the abrupt and unexplained jump from 0-indexed cardinal numbers to 1-indexed ordinal numbers. Absent a strong reason otherwise, it would seem clearly preferable to be consistent and avoid the obvious potential for confusion and ambiguity here, particularly in a technical explanation aimed at readers not already well-versed —as the old saying goes, there are two hard problems in programming: cache invalidation, naming things and off by one errors...

As such, I'd suggest

```suggestion
examined, then generation 1 is examined as well. With generation 2,
```